src/spatial/reaction_rates.jl

```diff
@@ -52,13 +52,20 @@ end
 
 update rates of all reactions in rxs at site
 """
-function update_rx_rates!(rx_rates, rxs, u, site)
+function update_rx_rates!(rx_rates::RxRates{F,M}, rxs, u, site) where {F, M <: MassActionJump}
     ma_jumps = rx_rates.ma_jumps
     @inbounds for rx in rxs
         set_rx_rate_at_site!(rx_rates, site, rx, evalrxrate((@view u[:,site]), rx, ma_jumps))
     end
 end
 
+function update_rx_rates!(rx_rates::RxRates{F,M}, rxs, u, site) where {F, M <: SpatialMassActionJump}
+    ma_jumps = rx_rates.ma_jumps
+    @inbounds for rx in rxs
+        set_rx_rate_at_site!(rx_rates, site, rx, evalrxrate((@view u[:,site]), rx, ma_jumps, site))
```

Comment:
We no longer need the view with the custom version of `evalrxrate` right? Couldn't we pass it the whole matrix `u` and just have it use `u[spec,site]` in the `SpatialMassActionJump` version?

Reply:
We can. What is the benefit of doing it that way?

Reply:
Avoids creating a `view`, which in the past had some overhead / performance cost (though it has been improved a lot over time). It just doesn't seem needed, so why keep it. This is the spatial version of `evalrxrate`, so it makes sense for that version to know that `u` is a matrix anyways.